### PR TITLE
Add 4844 containers to protobuf

### PR DESCRIPTION
**Pull request by terencechain**
**What type of PR is this?**

Spec (EIP4844)

**What does this PR do? Why is it needed?**

This PR adds the following containers to protobuf for EIP4844:
- `BlobsBundle`
    - Part of engine API
    - Currently defined under [here](https://github.com/ethereum/execution-apis/pull/197/files), but it should be merged shortly
- `SignedBeaconBlockAndBlobsSidecar`
    - https://github.com/ethereum/consensus-specs/blob/dev/specs/eip4844/p2p-interface.md#signedbeaconblockandblobssidecar 
- `BlobsSidecar`
    - https://github.com/ethereum/consensus-specs/blob/dev/specs/eip4844/p2p-interface.md#blobssidecar 

**Which issues(s) does this PR fix?**

Part of #11579 

**Other notes for review**

### CODE REVIEW

File: proto/engine/v1/execution_engine.proto [hunk 1]
```diff
@@ -142,3 +142,20 @@ message Withdrawal {
...
+	repeated bytes kzgs = 2 [(ethereum.eth.ext.ssz_size) = "?,48", (ethereum.eth.ext.ssz_max) = "16"];
+	// The blobs itself.
+	repeated Blob blobs = 3 [(ethereum.eth.ext.ssz_max)  = "16"];
+	// The aggregated proof of the blobs to enable fast validation for commitments and blobs.
+	bytes aggregated_proof = 4 [(ethereum.eth.ext.ssz_size) = "48"];
+}
+
+// Blob contains the data that is to be committed on chain.
+message Blob {
+	repeated bytes blob = 1 [(ethereum.eth.ext.ssz_size) = "4096,32"];
```
Comment (rauljordan) [inline proto/engine/v1/execution_engine.proto:160]: Are these going to be different for minimal config? if so we can probably use the dynamic pattern we used for the beacon state and blocks

**Review by rauljordan — DISMISSED** [review]
LGTM

File: proto/engine/v1/execution_engine.proto (hunk 1 above)
Comment (terencechain) [reply proto/engine/v1/execution_engine.proto:160]: Not as of now, we add can add them when minimal config is out

File: proto/engine/v1/execution_engine.proto [hunk 2]
```diff
@@ -142,3 +142,20 @@ message Withdrawal {
 	uint64 amount = 4;
 }
 
+// Blobs bundle is retrieved through engine-api from the execution layer client.
+// It consists of the necessary components for constructing a blobs sidecar object to gossip through p2p.
+message BlobsBundle {
+	// The block hash of the payload  which corresponds to the blobs.
+	bytes block_hash = 1 [(ethereum.eth.ext.ssz_size) = "32"];
+	// The KZG commitments of the blobs.
+	repeated bytes kzgs = 2 [(ethereum.eth.ext.ssz_size) = "?,48", (ethereum.eth.ext.ssz_max) = "16"];
```
Comment (prestonvanloon) [inline proto/engine/v1/execution_engine.proto:151]: Does this have to be named kzgs? How about calling it commitments? `kzgs` is hard type, say, and it's not very idiomatic.

**Review by prestonvanloon — CHANGES_REQUESTED** [review]

File: proto/engine/v1/execution_engine.proto (hunk 2 above)
Comment (terencechain) [reply proto/engine/v1/execution_engine.proto:151]: Yeah. I also like commitments better

Comment (terencechain) [reply proto/engine/v1/execution_engine.proto]: Discussed offline. `kzg_commitments` is 👑

**Review by prestonvanloon — APPROVED** [review]
LGTM

**Review by saolyn — APPROVED** [review]
LGTM

File: proto/engine/v1/execution_engine.proto [hunk 3]
```diff
@@ -142,3 +142,20 @@ message Withdrawal {
 	uint64 amount = 4;
 }
 
+// Blobs bundle is retrieved through engine-api from the execution layer client.
+// It consists of the necessary components for constructing a blobs sidecar object to gossip through p2p.
+message BlobsBundle {
+	// The block hash of the payload  which corresponds to the blobs.
```
Comment (rkapka) [inline proto/engine/v1/execution_engine.proto:148]:
```suggestion
	// The block hash of the payload which corresponds to the blobs.
```

File: proto/engine/v1/execution_engine.proto [hunk 4]
```diff
@@ -142,3 +142,20 @@ message Withdrawal {
...
+	// The blobs itself.
+	repeated Blob blobs = 3 [(ethereum.eth.ext.ssz_max)  = "16"];
+	// The aggregated proof of the blobs to enable fast validation for commitments and blobs.
+	bytes aggregated_proof = 4 [(ethereum.eth.ext.ssz_size) = "48"];
+}
+
+// Blob contains the data that is to be committed on chain.
+message Blob {
+	repeated bytes blob = 1 [(ethereum.eth.ext.ssz_size) = "4096,32"];
+}
```
Comment (rkapka) [inline proto/engine/v1/execution_engine.proto:161]: Can we rename `blob` to `data`? `Blob.Data` will be more readable than `Blob.Blob`.

**Review by rkapka — APPROVED** [review]
Approving because my commit dismissed previous reviews.

**Pull request merged**